page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataParser.java

```diff
@@ -76,70 +64,43 @@ private Node replaceArgumentPlaceholders(Node argument) throws DataException {
             argument = replaceDataTablePlaceholders(dataTable);
         } else if (argument instanceof DocString) {
             DocString docString = (DocString) argument;
-            argument = new DocString(docString.getLocation(), docString.getContentType(), replaceStepPlaceholders(docString.getContent()));
+            argument = new DocString(docString.getLocation(), docString.getContentType(), replaceDataPlaceholders(docString.getContent()));
 
         }
         return argument;
     }
 
-    private String replaceStepPlaceholders(String raw) throws DataException {
-        Pattern stepDataPattern = Pattern.compile(STEP_PARSE_REGEX);
+    private String replaceDataPlaceholders(String raw) throws DataException {
+        Pattern stepDataPattern = Pattern.compile(PATH_PARSE_REGEX);
         Matcher stepDataMatcher = stepDataPattern.matcher(raw);
         StringBuilder replacedStep = new StringBuilder(raw);
-        // Offset to handle diff between placeholder length and value length
-        int offset = 0;
 
         while (stepDataMatcher.find()) {
-            String collection = stepDataMatcher.group(2);
-            String value = stepDataMatcher.group(3);
+            String collection = stepDataMatcher.group(1);
+            String value = stepDataMatcher.group(2);
 
-            if (value == null) {
-                continue;
+            if (collection == null) {
+                parseDataTagValue(currentScenarioDataTagValue != null ? currentScenarioDataTagValue : featureDataTagValue);
```

Comment:
Format this one. I can't see the end of line on github

Reply:
It conforms to line length conventions. To see the full string on GitHub - change diff display type.